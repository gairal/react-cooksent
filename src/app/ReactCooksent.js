import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Cooksent from 'cooksent';

export default class ReactCooksent extends Component {
  static propTypes = {
    selector: PropTypes.string,
    btnSelector: PropTypes.string,
    title: PropTypes.string,
    description: PropTypes.string,
    btnLabel: PropTypes.string,
    onConsent: PropTypes.func,
    storageKey: PropTypes.string,
    force: PropTypes.bool,
    emtpy: PropTypes.func,
    show: PropTypes.func,
    hide: PropTypes.func,
  }

  static defaultProps = {
    selector: Cooksent.defaultOpts.selector,
    btnSelector: Cooksent.defaultOpts.btnSelector,
    title: Cooksent.defaultOpts.title,
    description: Cooksent.defaultOpts.description,
    btnLabel: Cooksent.defaultOpts.btnLabel,
    onConsent: Cooksent.defaultOpts.onConsent,
    storageKey: Cooksent.defaultOpts.storageKey,
    force: false,
    emtpy: () => {},
    show: () => {},
    hide: () => {},
  }

  constructor(props) {
    super(props);

    const {
      selector,
      btnSelector,
      title,
      description,
      btnLabel,
      onConsent,
      storageKey,
      force,
      emtpy,
      show,
      hide,
    } = props;

    this.force = force;

    // Expose the cooksent methods to the parent
    emtpy(this.emtpy.bind(this));
    show(this.show.bind(this));
    hide(this.hide.bind(this));

    this.cs = new Cooksent({
      selector,
      btnSelector,
      title,
      description,
      btnLabel,
      onConsent,
      storageKey,
    });
  }

  componentDidMount() {
    this.init();
  }

  init() {
    this.cs.init(this.force);
  }

  emtpy() {
    this.cs.storage.empty();
  }

  show() {
    this.cs.show();
  }

  hide() {
    this.cs.hide();
  }

  render() {
    return <React.Fragment />;
  }
}
