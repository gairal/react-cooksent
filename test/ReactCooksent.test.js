import React from 'react';
import renderer from 'react-test-renderer';
import ReactCooksent from '@/ReactCooksent';

describe('ReactCooksent', () => {
  let show;
  let empty;
  let hide;

  it('Renders correctly default', () => {
    const component = renderer.create(<ReactCooksent />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders correctly', () => {
    const component = renderer.create(<ReactCooksent
      title="custome title"
      description="custom description"
      btnLabel="Oki"
      storageKey="COOKSENT2"
      emtpy={(e) => { show = e; }}
      show={(e) => { empty = e; }}
      hide={(e) => { hide = e; }}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Can call methods', () => {
    show();
    empty();
    hide();
  });
});
