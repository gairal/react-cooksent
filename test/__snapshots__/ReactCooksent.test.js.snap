// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactCooksent Renders correctly 1`] = `null`;

exports[`ReactCooksent Renders correctly default 1`] = `null`;
